job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java

```diff
@@ -1173,6 +1177,148 @@ public void testJobDeletionWithLabels() throws Exception {
         JobServiceDatabaseUtil.assertJobLabelRowsDoNotExist(job1Id);
     }
 
+    @Test
+    public void testPauseWaitingJobIsSuccessful() throws Exception {
+        final String jobId = generateJobId();
+        createJob(jobId, true);
+        waitUntilJobStatusIs(JobStatus.Waiting, jobId);
+        pauseJob(jobId);
+        waitUntilJobStatusIs(JobStatus.Paused, jobId);
+    }
+
+    @Test
+    public void testPausePausedJobIsSuccessful() throws Exception {
+        final String jobId = generateJobId();
+        createJob(jobId, true);
+        waitUntilJobStatusIs(JobStatus.Waiting, jobId);
+        pauseJob(jobId);
+        waitUntilJobStatusIs(JobStatus.Paused, jobId);
+        pauseJob(jobId);
+        waitUntilJobStatusIs(JobStatus.Paused, jobId);
+    }
+
+    @Test
+    public void testPauseCompletedJobIsNotAllowed() throws Exception {
+        exampleWorkerMessageOutQueue = null; // Setting this to null means the job completes rather than being stuck in a 'Waiting' state
+        final String jobId = generateJobId();
+        createJob(jobId, true);
+        waitUntilJobCompletes(jobId);
+        try {
+            pauseJob(jobId);
+        } catch (final ApiException e) {
+            Assert.assertEquals(e.getCode(), 400, "Unexpected HTTP response code");
+            Assert.assertEquals(e.getMessage(),
+                                "{\"message\":\"ERROR: job_id {" + jobId + "} cannot be paused as it has a status of {Completed}. "
+                                + "Only jobs with a status of Active or Waiting can be paused.\\n  "
+                                + "Where: PL/pgSQL function pause_job(character varying,character varying) line 23 at RAISE\"}",
```

Comment:
It's not a great idea to test the exact error message like this.  In this case even adding a blank line to the function would cause the tests to break, nevermind how the message might change with a bigger change like a PostgreSQL upgrade.

Reply:
Updated to use `contains` rather than `equals` to just check the error message for the subset of text of interest rather than the entire thing.